subscription.go

```diff
@@ -162,19 +162,21 @@ func (r *marathonClient) registerCallbackSubscription() error {
 	return nil
 }
 
-func (r *marathonClient) registerSSESubscription() error {
-	// Prevent multiple SSE subscriptions
+func (r *marathonClient) registerSSESubscriptionOnce() error {
 	if r.subscribedToSSE {
 		return nil
 	}
 
-	request, _, err := r.buildAPIRequest("GET", marathonAPIEventStream, nil)
-	if err != nil {
+	if err := r.registerSSESubscription(); err != nil {
 		return err
 	}
 
-	// Try to connect to stream, reusing the http client settings
-	stream, err := eventsource.SubscribeWith("", r.config.HTTPClient, request)
+	r.subscribedToSSE = true
+	return nil
+}
+
+func (r *marathonClient) registerSSESubscription() error {
```

Comment:
How will this method behave when `RemoveEventsListener` is called? Wouldn't we need to inject some stop channel / context that we use a signal to shut down?